packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -161,27 +199,48 @@ export class GitQuickOpenService {
 
     private async getRemotes(): Promise<string[]> {
         const repository = this.getRepository();
-        return repository ? this.git.remote(repository) : [];
+        try {
+            return repository ? await this.git.remote(repository) : [];
+        } catch (error) {
+            this.logError(error);
+            return [];
+        }
     }
 
     private async getBranches(): Promise<Branch[]> {
         const repository = this.getRepository();
         if (!repository) {
             return [];
         }
-        const [local, remote] = await Promise.all([
-            this.git.branch(repository, { type: 'local' }),
-            this.git.branch(repository, { type: 'remote' })
-        ]);
-        return [...local, ...remote];
+        try {
+            const [local, remote] = await Promise.all([
+                this.git.branch(repository, { type: 'local' }),
+                this.git.branch(repository, { type: 'remote' })
+            ]);
+            return [...local, ...remote];
+        } catch (error) {
+            this.logError(error);
+            return [];
+        }
     }
 
     private async getCurrentBranch(): Promise<Branch | undefined> {
         const repository = this.getRepository();
         if (!repository) {
             return undefined;
         }
-        return await this.git.branch(repository, { type: 'current' });
+        try {
+            return await this.git.branch(repository, { type: 'current' });
+        } catch (error) {
+            this.logError(error);
+            return undefined;
+        }
+    }
+
+    // tslint:disable-next-line:no-any
+    private logError(error: any): void {
```

Comment:
This should be `error: Error | string` ?
since this.messageService.error is taking a string or undefined 
Or we should have to way to check if it's a string before using that method ?

Reply:
OK, I will update this as part of another PR. Thanks for the reminder.

Reply:
Why didn't I get an error when I tried to call `messageService.error` with type of `any`?

Reply:
I think it's because it can find a path that works:

```
const message = error instanceof Error ? error.message : error;
+        this.messageService.error(message);
```
if error is Error then you have error.message and error.message is a string so it's ok.

It doesn't check that all the branches work. only that one can work.

I had this issue somewhere else too.

Reply:
Then this is a bug in tsc, isn't it?

Reply:
Not sure, maybe more a missing feature :)

Reply:
Possibly checking all the possible branches is a problem that can grow exponentially and take way to much ressources

Reply:
it is ok because `message` is `string | any`. `any`disables type checks.